src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -66,13 +66,15 @@ private[xml] object StaxXmlParser extends Serializable {
       // create a row even if no corrupt record column is present
       options.parseMode match {
         case FailFastMode =>
-          throw new RuntimeException(
+          throw new IllegalArgumentException(
```

Comment:
I also changed these to a more specific exception; anyone catching `RuntimeException` still catches this, but can maybe avoid catching every unchecked exception in the process.